node/core/dispute-coordinator/src/initialized.rs

```diff
@@ -1003,6 +1003,13 @@ impl Initialized {
 
 		gum::trace!(target: LOG_TARGET, ?candidate_hash, ?session, "Loaded votes");
 
+		let controlled_indices = env.controlled_indices();
+		let own_statements = statements
+			.iter()
+			.filter(|(_, index)| controlled_indices.contains(index))
```

Comment:
We should also filter on `candidate_hash` otherwise we'd later match wrongly on different candidate own statemets.

Reply:
Or is it guaranteed that all statements here are for the `candidate_hash` in question ?

Reply:
I'm not sure. But I added the filter just in case.

Reply:
Afaik they all are for [the candidate hash passed to the function](https://github.com/paritytech/polkadot/pull/7404/files/e5e0280a52fbd1542aaa80b42732e7f460a56f6a#diff-4394711f4990be21fdbeb6e1dad6f4a0993f5f024382e0e7ac5534411b377695R921). But better safe than sorry.